modules/wav_loader/register_types.cpp

```diff
@@ -0,0 +1,424 @@
+	}
+
+	int import_loop_mode = p_options["edit/loop_mode"];
+	int loop_begin = 0;
+	int loop_end = 0;
+	AudioStreamWAV::LoopMode loop_mode = AudioStreamWAV::LoopMode::LOOP_DISABLED;
+
+	Dictionary tag_dictionary;
+	for (uint32_t meta = 0; meta < p_wav.metadataCount; ++meta) {
+		drwav_metadata md = p_wav.pMetadata[meta];
+		if (import_loop_mode == 0 && md.type == drwav_metadata_type_smpl && md.data.smpl.sampleLoopCount > 0) { // Loop info
+			drwav_smpl_loop loop = md.data.smpl.pLoops[0];
+			loop_mode = (AudioStreamWAV::LoopMode)(loop.type + 1);
+			loop_begin = loop.firstSampleOffset;
+			loop_end = loop.lastSampleOffset;
+		} else if (md.type & drwav_metadata_type_list_all_info_strings) { // Metadata tags
+			// Name tags accordingly.
+			String tag;
+			switch (md.type) {
+				case drwav_metadata_type_list_info_software:
+					tag = "software";
+					break;
+				case drwav_metadata_type_list_info_copyright:
+					tag = "copyright";
+					break;
+				case drwav_metadata_type_list_info_title:
+					tag = "title";
+					break;
+				case drwav_metadata_type_list_info_artist:
+					tag = "artist";
+					break;
+				case drwav_metadata_type_list_info_comment:
+					tag = "comments";
+					break;
+				case drwav_metadata_type_list_info_date:
+					tag = "date";
+					break;
+				case drwav_metadata_type_list_info_genre:
+					tag = "genre";
+					break;
+				case drwav_metadata_type_list_info_album:
+					tag = "album";
+					break;
+				case drwav_metadata_type_list_info_tracknumber:
+					tag = "tracknumber";
+					break;
+				default: // Unsupported or unknown tag
+					break;
+			}
+
+			// The data is always an ASCII string. ASCII is a subset of UTF-8.
+			String tag_value;
+			tag_value.append_utf8(md.data.infoText.pString, md.data.infoText.stringLength);
+
+			tag_dictionary[tag] = tag_value;
+		}
+	}
+
+	// dr_wav doesn't check for incorrect data chunk sizes, so we do it here.
+	uint8_t bytes_per_sample = drwav_get_bytes_per_pcm_frame(&p_wav);
+	uint64_t expected_data_size = frames * bytes_per_sample;
+	uint64_t actual_data_size = data_length - p_wav.dataChunkDataPos;
+	if (actual_data_size < expected_data_size) {
+		frames = actual_data_size / bytes_per_sample;
+		WARN_PRINT("Data chunk size is smaller than expected. Proceeding with actual data size.");
+	}
+
+	Vector<float> data;
+	data.resize(frames * format_channels);
+	drwav_read_pcm_frames_f32(&p_wav, frames, data.ptrw());
+	drwav_uninit(&p_wav);
+
+	// STEP 2, APPLY CONVERSIONS
+
+	bool is16 = format_bits != 8;
+	int rate = format_freq;
+
+	/*
+	print_line("Input Sample: ");
+	print_line("\tframes: " + itos(frames));
+	print_line("\tformat_channels: " + itos(format_channels));
+	print_line("\t16bits: " + itos(is16));
+	print_line("\trate: " + itos(rate));
+	print_line("\tloop: " + itos(loop));
+	print_line("\tloop begin: " + itos(loop_begin));
+	print_line("\tloop end: " + itos(loop_end));
+	*/
```

Comment:
Remove commented out code (or if it's meant to be toggled quickly, put it behind `#if 0` instead):

```suggestion
```